### ISSUE TITLE
Custom path input for resnet download

### ISSUE THREAD

**Issue by Smkmw** [DeepLabCut/DeepLabCut#2162]
I'm working on a shared cluster, and am running DLC within a docker container environment (2.2.1.1-base-cuda11.0.3-runtime-ubuntu18.04) with singularity.

On the cluster I only have certain permissions.
The problem now arises when creating a trainings dataset, the resnet-image gets downloaded into a directory in which I dont have any permissions. This then gives an error, and no training dataset is created.

A solution would be the possibility to give a custom path in which to download the resnet. (found this suggestion multiple times)
I found that I should edit the following code:
https://github.com/DeepLabCut/DeepLabCut/blob/7c1c8719426439a57fea8c07b37afee222d83fc7/deeplabcut/utils/auxfun_models.py#L125
and
https://github.com/DeepLabCut/DeepLabCut/blob/7c1c8719426439a57fea8c07b37afee222d83fc7/deeplabcut/utils/auxfun_models.py#L26

However, I have no idea how I would change the code in the docker image. As I found this as solution for usage of DLC not in a docker container.

A work-around is creating the trainingdataset on my "labelling pc" which has no GPU. And then moving everything to the cluster, manually download a resnet, and set this resnet as initial weight. AND fix all paths in the config.yaml and both pose_cfg.yaml because the system at which I create the training dataset runs on windows, and where I train runs linux (I have to change al the \ to / ).

This is fine for training 1 time, but when training multiple networks and making multiple projects, this gets quite time consuming.
So an option for an input for specifying a custom path for the resnet download would be nice. Especially for the use of the dockers in shared systems where permission problems may arise.

**Comment by MakeSometh1ngWonderful** [comment]
Potentially you could:

1. [Install a copy](https://deeplabcut.github.io/DeepLabCut/docs/installation.html) of DLC on your local machine.
2. As you have suggested, edit the lines in [DeepLabCut/deeplabcut/utils/auxfun_models.py](https://github.com/DeepLabCut/DeepLabCut/blob/7c1c8719426439a57fea8c07b37afee222d83fc7/deeplabcut/utils/auxfun_models.py#L125)
 Line 125 in [7c1c871](https://github.com/DeepLabCut/DeepLabCut/commit/7c1c8719426439a57fea8c07b37afee222d83fc7)
3. Create a new docker [image and container](https://github.com/DeepLabCut/Docker4DeepLabCut2.0#step-by-step-instructions-for-creating-the-docker-environment) from your custom copy of DLC.
4. Push the new container to your cluster.

This would allow you to create and maintain custom containers suited to your needs.

**Comment by Smkmw** [comment]
Hi element, you recon it would work as my local machine runs windows without any GPU, and my cluster runs on Linux with GPU support. So my installs are quite different I guess? Not sure if that would give any problems

**Comment by MakeSometh1ngWonderful** [comment]
Sure, docker has some solutions for this. You will need [docker for windows](https://docs.docker.com/desktop/install/windows-install/#start-docker-for-windows). 
Then, you could do something like the following:
`docker pull ubuntu:<version>` 
(Here is a link to their [ubuntu image](https://hub.docker.com/_/ubuntu))
Use "docker images" to find the ID of the image that the previous command pulls.
Then, create the new linux container with
`docker run -i -t <image ID> /bin/bash`
This creates a container and launches a bash prompt. From there, I would follow the DLC instructions for install and set it up as if it was running on the cluster, in effect, editing the lines for the 'dlc_path'. Then, create an [image from this container](https://thenewstack.io/tutorial-create-a-docker-image-from-a-running-container/) and push this to the cluster. 
Unfortunately, it sounds like you will have to create the image locally and test it on the cluster due to your specific combination of permission and GPU constraints.

**Comment by AlexEMG** [comment]
Note, you can also download it the resnet weights to any folder of interest, by running: 

`
curl http://download.tensorflow.org/models/resnet_v1_50_2016_08_28.tar.gz | tar xvz
`

Then in the `pose_cfg.yaml` you can just set the path to that file when you train the model (see Box 2 in the guide: https://deeplabcut.github.io/DeepLabCut/docs/standardDeepLabCut_UserGuide.html#f-create-training-dataset-s). You can even set the path in the default_pose_config file, and then for any project on that machine, for which you would create a training set, it will link to the correct resnet weights.

**Comment by Smkmw** [comment]
Hey @AlexEMG , My problem was that training dataset does not get created when the download of the resnet fails. So the pose_cfg.yaml isn't created either for setting the paths correctly to a manually downloaded resnet.
That why I created a training dataset on another machine, but theres a couple of paths that have to be corrected, as I created on a windows system and train on a linux system (I have to change all the "\" to "/" because the paths get weird otherwise.

And if I understand correctly, I can change the default pose config, where would I find that? Cant seem to find it somewhere
